src/OpenAL/OpenToolkit.OpenAL/Native/ApiBase.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text;
+using OpenToolkit.Core.Loader;
+
+namespace OpenToolkit.Audio.OpenAL
+{
+    /// <summary>
+    /// Provides a base for ApiContext so that it can register dll intercepts.
+    /// </summary>
+    public abstract class ApiBase
+    {
+        private static readonly Dictionary<string, IPlatformLibraryNameContainer> DllIntercepts = new Dictionary<string, IPlatformLibraryNameContainer>();
+
+        static ApiBase()
+        {
+            NativeLibrary.SetDllImportResolver(typeof(ApiBase).Assembly, ImportResolver);
+        }
+
+        private static IntPtr ImportResolver(string libraryName, Assembly assembly, DllImportSearchPath? searchPath)
+        {
+            IntPtr libHandle = IntPtr.Zero;
+            if (DllIntercepts.TryGetValue(libraryName, out var libraryNameContainer))
+            {
+                string libName = libraryNameContainer.GetLibraryName();
+
+                if (NativeLibrary.TryLoad(libName, assembly, searchPath, out libHandle) == false)
+                {
+                    throw new DllNotFoundException($"Could not load the dll '{libName}' (this load is intercepted, specified in DllImport as '{libraryName}').");
```

Comment:
I'm not sure if this is a good error message for this. Suggestions welcome! (if this isn't a super good message already 🙂)

Reply:
I think its alright. Can you actually try  it out if it also works as intended here? Maybe have

Reply:
This is how the error message looks when I specify the wrong name in `OpenALLibraryNameContainer`.
![image](https://user-images.githubusercontent.com/3778357/75299339-13633080-5835-11ea-8c03-7cb8f0032165.png)